### PR TITLE
support for llvm ASAN use -DLLVM_SANITIZE_ADDRESS=1

### PR DESCRIPTION
**Pull request by PDoakORNL**
MSAN is also included in theory but I have not gotten a complete set of instrumented libraries built yet and the blacklist/whitelist feature seems buggy. So it is not useful yet in my opinion.

### CODE REVIEW

File: CMake/ClangCompilers.cmake [hunk 1]
```diff
@@ -32,6 +32,8 @@ SET( CMAKE_C_FLAGS_RELEASE     "${CMAKE_C_FLAGS_RELEASE} -ffast-math" )
 SET( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -ffast-math" )
 SET( CMAKE_C_FLAGS_RELWITHDEBINFO     "${CMAKE_C_FLAGS_RELWITHDEBINFO} -ffast-math" )
 SET( CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -ffast-math" )
+SET( CMAKE_C_FLAGS_DEBUG     "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer -fstandalone-debug" )
```
Comment (ye-luo) [inline CMake/ClangCompilers.cmake]: Please a comment and separates from setting optimization flags.
No BOOST_HANA

**Comment by ye-luo** [comment]
Please mention in the manual even just a line how to enable this feature.

**Comment by prckent** [comment]
Maybe test drive on a few systems before putting in the manual? Key for me would be a recipe that I could use on a regular Linux workstation e.g. oxygen.

**Comment by PDoakORNL** [comment]
These flags will work fine if you are building with the default llvm7 from spack plus everything else we use.

File: manual/installation.tex [hunk 2]
```diff
@@ -324,6 +324,15 @@ \subsection{Configuration options}
                        e.g. "-n", "-np", etc.
 \end{verbatim}
 
+\item LLVM/Clang Developer Options\\
```
Comment (ye-luo) [inline manual/installation.tex]: It is better to put them in the tool section near profiling tools.

Comment (prckent) [reply manual/installation.tex]: Ye - this is a cmake config option and needs to go here.  This section should be comprehensive. I suggest merging now and adding an additional set of notes in the tools section after we have some experience.

Comment (ye-luo) [reply manual/installation.tex]: This is a compiler or tool specific option.
Listing an entry here is fine but the following bulk description must be moved to the tools section.

We were not listing CMake options for profiling tools. If the policy is having all the CMake options listed here. We need to list all of them.

Comment (ye-luo) [reply manual/installation.tex]: I'd like to have an entry as brief as possible. Even not listing the two options here. Just put a ref to the tool section.

**Comment by ye-luo** [comment]
Okay to test

**Pull request merged**